domain/relation/state/relation.go

```diff
@@ -1072,33 +1093,187 @@ WHERE  name = $getUnit.name
 			return errors.Capture(err)
 		}
 
-		// Insert or replace relation unit settings
-		err = st.updateUnitSettings(ctx, tx, relationUnitUUID, settings)
+		// Set the relation unit settings.
+		err = st.setRelationUnitSettings(ctx, tx, relationUnitUUID, settings)
 		if err != nil {
+			return errors.Errorf("setting relation unit settings: %w", err)
+		}
+
+		return nil
+	})
+	return errors.Capture(err)
+}
+
+// CreateSubordinateParams checks if there is a subordinate application
+// related to the principal unit that needs a subordinate unit created.
+//
+// In the case that all the following hold, parameters for creating a
+// subordinate unit will be returned:
+//   - The unit and relation are alive.
+//   - The unit is in the relation.
+//   - The relation is container scoped.
+//   - The relation relates a subordinate application to a principal application.
+//   - The unit is on the principal application.
+//   - The unit does not already have a subordinate unit from the subordinate app.
+//
+// If any of these conditions do not hold, nil, or an error will be returned.
+//
+// The following error types can be expected to be returned:
+//   - [relationerrors.RelationNotFound] if the relation is not found.
+//   - [relationerrors.RelationNotAlive] if the relation is not alive.
+//   - [relationerrors.RelationUnitNotFound] if the relation unit cannot be found.
+//   - [relationerrors.UnitNotAlive] if the principal unit is not alive.
+func (st *State) CreateSubordinateParams(
+	ctx context.Context,
+	relationUUID corerelation.UUID,
+	principalUnitName unit.Name,
+) (*application.ID, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	var result *application.ID
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+
+		// Check the relation exists.
+		if relationFound, err := st.checkExistsByUUID(ctx, tx, "relation", relationUUID.String()); err != nil {
 			return errors.Capture(err)
+		} else if !relationFound {
+			return relationerrors.RelationNotFound
+		}
+
+		// Check the relation is alive.
+		if alive, err := st.checkLife(ctx, tx, "relation", relationUUID.String(), life.IsAlive); err != nil {
+			return errors.Errorf("getting relation life: %w", err)
+		} else if !alive {
+			return relationerrors.RelationNotAlive
```

Comment:
todo: remove me, these checks are superfluous as the error types aren't checked. `getRelationUnit` does it all.